docs/src/main/asciidoc/spring-cloud-vault.adoc

```diff
@@ -611,8 +612,47 @@ NOTE: Enabling multiple JDBC-compliant databases will generate credentials
 and store them by default in the same property keys hence property names for
 JDBC secrets need to be configured separately.
 
+[[vault.config.backends.database]]
+=== Database
+
+Spring Cloud Vault can obtain credentials for any Database listed at
+https://www.vaultproject.io/api/secret/databases/index.html.
+The integration can be enabled by setting
+`spring.cloud.vault.database.enabled=true` (default `false`) and
+providing the role name with `spring.cloud.vault.database.role=…`.
+
+Username and password are stored in `spring.datasource.username`
+and `spring.datasource.password` so using Spring Boot will
+pick up the generated credentials without further configuration.
+You can configure the property names by setting
+`spring.cloud.vault.database.username-property` and
+`spring.cloud.vault.database.password-property`.
+
+====
+[source,yaml]
+----
+spring.cloud.vault:
+    database:
+        enabled: true
+        role: readonly
+        backend: database
+        username-property: spring.datasource.username
+        password-property: spring.datasource.username
+----
+====
+
+* `enabled` setting this value to `true` enables the Database backend config usage
+* `role` sets the role name of the Database role definition
+* `backend` sets the path of the Database mount to use
+* `username-property` sets the property name in which the Database username is stored
+* `password-property` sets the property name in which the Database password is stored
+
+See also: https://www.vaultproject.io/docs/secrets/databases/index.html[Vault Documentation: Database Secrets backend]
+
+
 [[vault.config.backends.cassandra]]
 === Apache Cassandra
+This backend has been deprecated and might be removed from Vault soon.
```

Comment:
We should not deprecate this one because it maps to different properties. It's common to use a single JDBC database along with several NoSQL stores. Removing/deprecating removes the ability to plug into multiple databases with Spring Boot and Vault.

It would rather make sense to hint towards the database backend.

Reply:
I guess this makes sense. I'll add in the documentation that the 'database' backend should be used but it is a good idea to mount it as 'cassandra' in this case.